ls1intum/Artemis
# Enhancement/exam mode/client/improve test coverage exam participation

### Checklist


### Motivation and Context


according to the code coverage report the coverage in the client side is still not enough and an improvement is needed for the services. This pr resolve the low coverage of exam-participation.service.ts and contribute to #2081 


### Description

using Jest to add test for all the public methods in exam-participation.service.ts  which imitate the RESTCall, local-storage saving, session-storage saving by mock object

### Steps for Testing


1. Log in to Bamboo
2. Navigate to this branch
3. see the code coverage report from client side or re run to get the newest code coverage report

### Test Coverage

*exam-participation.service.ts from  21.05% in develop to 92.11% in this branch

### Screenshots


in develop:
![Screenshot 2020-10-07 at 17 18 34](https://user-images.githubusercontent.com/48210790/95351150-2fe24380-08c1-11eb-814f-2be50fbe01f9.png)
in this branch:
![Screenshot 2020-10-07 at 17 20 40](https://user-images.githubusercontent.com/48210790/95351398-7899fc80-08c1-11eb-8ecd-86ff3e290c9b.png)
